recipes/libcap/all/conanfile.py

```diff
@@ -53,6 +53,12 @@ def _configure_autotools(self):
         self._autotools_env["DESTDIR"] = self.package_folder
         self._autotools_env["prefix"] = "/"
         self._autotools_env["lib"] = "lib"
+
+        if tools.cross_building(self.settings):
+            native_cc = tools.which("cc")
+            self.output.info("Using native compiler '{}'".format(native_cc))
+            self._autotools_env["BUILD_CC"] = native_cc
+
```

Comment:
Why are you setting `BUILD_CC`?
How about this?
```suggestion
        self._autotools_env["lib"] = "lib"
        if tools.get_env("CC"):
            self._autotools_env["CC"] = tools.get_env("CC")
        if tools.get_env("AR"):
            self._autotools_env["AR"] = tools.get_env("AR")
        if tools.get_env("RANLIB"):
            self._autotools_env["RANLIB"] = tools.get_env("RANLIB")
```

Reply:
libcap is creating and using [some code generator](https://git.kernel.org/pub/scm/libs/libcap/libcap.git/tree/libcap/Makefile?h=libcap-2.45#n68) during build process. So in cross building scenario libcap requires two compilers:
* `CC` - cross-compiler to create libraries for a target architecture (arm-linux-gnueabihf-gcc in my case)
* `BUILD_CC` - native-compiler to create tools for a build host architecture (x86_64-linux-gnu-gcc in my case)

`self._autotools_env["CC"] = tools.get_env("CC")` is not necessary because CC is already in the environment variables. It will just redefine `CC` with the same value.

Reply:
You're correct.
Is it btw guaranteed that `cc` is a native compiler?

Reply:
No. It is not guaranteed. But most of systems is managing it as a symlink to default compiler, so it should be not very bad assumption that it is a native compiler.
Some projects use something like this in the build files: [`AC_CHECK_PROGS(BUILD_CC, gcc cc)`](https://git.kernel.org/pub/scm/fs/xfs/xfsprogs-dev.git/tree/configure.ac#n18)(just checks for gcc or cc commands)

But probably i shouldn't override `BUILD_CC` unconditionally. It would be better to make
```
if not tools.get_env("BUILD_CC"):
    self._autotools_env["BUILD_CC"] = native_cc
```

Reply:
I guess these kind of hacks are needed until conan has good cross building support (for multiple toolchains).

Reply:
btw, does the `Makefile` allow to override `BUILD_CC`?

Reply:
Yes it [does](https://git.kernel.org/pub/scm/libs/libcap/libcap.git/tree/Make.Rules?h=libcap-2.45#n61).

Reply:
Thanks @bobrofon for checking!

Reply:
@jgsogo 
This is my bimonthly question to you asking whether there is an update on `BUILD_CC`/`BUILD_CXX` for cross building? :smile: 
Like mapping `CC/CXX` from build requirements to `CC_BUILD`/`CXX_BUILD`.
Or getting access to these variables in some python object.